### PR TITLE
Add test and docs for nested_components property

### PR DESCRIPTION
**Pull request by CassianoMontibeller**

### CODE REVIEW

File: tests/Components/test_BaseComponent.py [hunk 1]
```diff
@@ -61,3 +61,23 @@ def test_angle_of_attack(freebody_component, attached_component):
         math.degrees(freebody_component.angle_of_attack), -5.0, decimal=1)
     npt.testing.assert_almost_equal(
         math.degrees(attached_component.angle_of_attack), 4.0, decimal=1)
+
+
+def test_nested_components(base_component, freebody_component, attached_component):
+    base_component.append_child(freebody_component)
```
Comment (rafaellehmkuhl) [inline tests/Components/test_BaseComponent.py:67]: Same thing here.

File: tests/Components/test_BaseComponent.py [hunk 2]
```diff
@@ -19,7 +19,7 @@ def base_component():
 @pytest.fixture
 def freebody_component():
     freebody_component = FreeBody(
-        name='component',
+        name='freebody_component',
```
Comment (rafaellehmkuhl) [inline tests/Components/test_BaseComponent.py:22]: Just remove this modification as it doens't have to do with this change.

File: docs/Components/BaseComponent.md [hunk 3]
```diff
@@ -63,3 +63,11 @@ print(math.degrees(freebody_component.angle_of_attack))
...
 >>> 4.0
 ```
+## nested_components property
+This property returns a dictionary with the hierarchical relationship of each component
+```python
+base_component.append_child(attached_component)
+
+print(base_component.nested_components)
+>>>{'component': BaseComponent(name='component', type='generic_component', mass=3.4, children={'wing1': BaseComponent(name='wing1', type='wing', mass=1.1, children={}, external_forces={}, external_moments={}), 'attached_component': AttachedComponent(name='attached_component', type='generic_attached_component', mass=1.4, children={}, external_forces={}, external_moments={}, parent=None, relative_position=<Vector2 (-0.4, 0.1)>, relative_angle=0.15707963267948966, actuation_angle=0.0)}, external_forces={}, external_moments={}), 'wing1': BaseComponent(name='wing1', type='wing', mass=1.1, children={}, external_forces={}, external_moments={}), 'attached_component': AttachedComponent(name='attached_component', type='generic_attached_component', mass=1.4, children={}, external_forces={}, external_moments={}, parent=None, relative_position=<Vector2 (-0.4, 0.1)>, relative_angle=0.15707963267948966, actuation_angle=0.0)}
```
Comment (rafaellehmkuhl) [inline docs/Components/BaseComponent.md:72]: You already append a child earlier, so no need to append another one.

File: tests/Components/test_BaseComponent.py [hunk 4]
```diff
@@ -19,7 +19,7 @@ def base_component():
 @pytest.fixture
 def freebody_component():
     freebody_component = FreeBody(
-        name='component',
+        name='componente',
```
Comment (rafaellehmkuhl) [inline tests/Components/test_BaseComponent.py:22]: Still changing hehe

File: docs/Components/BaseComponent.md [hunk 5]
```diff
@@ -63,3 +63,9 @@ print(math.degrees(freebody_component.angle_of_attack))
 print(math.degrees(attached_component.angle_of_attack))
 >>> 4.0
 ```
+## nested_components property
+This property returns a dictionary with the hierarchical relationship of each component
+```python
+print(base_component.nested_components)
+>>> {'component': BaseComponent(name='component', type='generic_component', mass=3.4, children={'wing1': BaseComponent(name='wing1', type='wing', mass=1.1, children={}, external_forces={}, external_moments={})}, external_forces={}, external_moments={}), 'wing1': BaseComponent(name='wing1', type='wing', mass=1.1, children={}, external_forces={}, external_moments={})}
+```
```
Comment (rafaellehmkuhl) [inline docs/Components/BaseComponent.md:71]: Nice!

**Comment by rafaellehmkuhl** [comment]
All good.
Just squash the commits (reset mixed) and rebase and we are ready to merge.

**Review by rafaellehmkuhl — APPROVED** [review]

**Review by joaofoes — APPROVED** [review]

**Pull request merged**